### ISSUE TITLE
datastore: PropertyLoadSaver ignored with nil map types

### ISSUE THREAD

**Issue by yuryu** [googleapis/google-cloud-go#2611]
**Client**

Datastore

**Environment**

Firestore on Datastore mode
MacOS 10.15.5

**Go Environment**

```
$ go version
go version go1.14.6 darwin/amd64
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/yuryu/Library/Caches/go-build"
GOENV="/Users/yuryu/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/yuryu/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/yuryu/src/triton/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/kt/8436st1j7jb_pmp8wwcm7z38009x4l/T/go-build503312083=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

**Code**

```go
package main

import(
	"fmt"
	"cloud.google.com/go/datastore"
)

type Map map[int]int

func (_ *Map) Load(_ []datastore.Property) error {
	return nil
}

func (_ *Map) Save() ([]datastore.Property, error) {
	return []datastore.Property{}, nil
}

type Struct struct {
	Map Map
}

func main() {
	s := &Struct{}
	fmt.Println(datastore.SaveStruct(s))
}
```

**Expected behavior**

The datastore client uses Save() and Load() methods defined to load and save PropertyMap.
The [document](https://pkg.go.dev/cloud.google.com/go/datastore?tab=doc#hdr-The_PropertyLoadSaver_Interface) says 
> An entity's contents can also be represented by any type that implements the PropertyLoadSaver interface. This type may be a struct pointer, but it does not have to be.

**Actual behavior**

"[] datastore: unsupported struct field type: main.Map" error is returned if Map is nil. If I change main() to
```go
func main() {
	s := &Struct{make(Map)}
	fmt.Println(datastore.SaveStruct(s))
}
```
the error disappears.

**Comment by tritone** [comment]
Thanks for your report @yuryu .

@IlyaFaer could you take a look at this one? Seems very similar to the issue that you fixed in #2613 .

**Comment by IlyaFaer** [comment]
@tritone, sure
